File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	var n int
	fmt.Fscanf(reader, "%d\n", &n)
	var res = make([]int, n)
	var numbers = make([]int, n)
	var max int
	for i := 0; i < n; i++ {
		fmt.Fscanf(reader, "%d ", &numbers[i])
	}
	for i := 0; i < n; i++ {
		for j := 0; j < i; j++ {
			if numbers[i] > numbers[j] && res[i] < res[j] {
				res[i] = res[j]
			}
		}
		res[i] += 1
	}
	for _, v := range res {
		if v > max {
			max = v
		}
	}
	fmt.Fprintf(writer, "%d\n", max)
}

File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
N = int(input())
res = [0 for _ in range(N)]
numbers = list(map(int, input().split()))
for i in range(N):
    for j in range(i):
        if numbers[i]>numbers[j] and res[i]<res[j]:
            res[i]=res[j]
    res[i]+=1
print(max(res))
